core/adapters/bridge.go

```diff
@@ -31,59 +31,82 @@ func (ba *Bridge) Perform(input models.RunResult, store *store.Store) models.Run
 	} else if input.Status.PendingBridge() {
 		return resumeBridge(input)
 	}
-	ba.handleNewRun(&input, store.Config.BridgeResponseURL())
-	return input
+	return ba.handleNewRun(input, store.Config.BridgeResponseURL())
 }
 
 func resumeBridge(input models.RunResult) models.RunResult {
 	input.Status = models.RunStatusInProgress
 	return input
 }
 
-func (ba *Bridge) handleNewRun(input *models.RunResult, bridgeResponseURL *url.URL) {
+func (ba *Bridge) handleNewRun(input models.RunResult, bridgeResponseURL *url.URL) models.RunResult {
```

Comment:
We've tried to move away from this sort of pseudo-functional style because it's not very "go" like, and introducing it here mixes a lot of different styles, making it confusing to reason about.

Large objects like `models.RunResult` should generally be passed in as pointers in go.